internal/operands/vm-delete-protection/reconcile.go

```diff
@@ -0,0 +1,82 @@
+package vm_delete_protection
+
+import (
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Define RBAC rules needed by this operand:
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingadmissionpolicies,verbs=get;list;create;watch;update
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=validatingadmissionpolicybindings,verbs=get;list;create;watch;update
+
+const (
+	operandName                                     = "vm-delete-protection"
+	operandComponent                                = common.AppComponetVMDeletionProtection
+	virtualMachineDeleteProtectionPolicyName        = "ssp-vm-deletion-protection-policy"
+	virtualMachineDeleteProtectionPolicyBindingName = "ssp-vm-deletion-protection-binding"
+)
+
+func init() {
+	utilruntime.Must(admissionregistrationv1.AddToScheme(common.Scheme))
+}
+
+type VMDeleteProtection struct{}
+
+var _ operands.Operand = &VMDeleteProtection{}
+
+func New() operands.Operand { return &VMDeleteProtection{} }
+
+func (v *VMDeleteProtection) WatchTypes() []operands.WatchType {
+	return nil
+}
+
+func (v *VMDeleteProtection) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &admissionregistrationv1.ValidatingAdmissionPolicy{}},
+		{Object: &admissionregistrationv1.ValidatingAdmissionPolicyBinding{}},
+	}
+}
+
+func (v *VMDeleteProtection) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	return common.CollectResourceStatus(request,
+		reconcileVAP,
+		reconcileVAPB,
+	)
+}
+
+func (v *VMDeleteProtection) Cleanup(request *common.Request) ([]common.CleanupResult, error) {
+	return common.DeleteAll(request)
```

Comment:
This function takes as parameter objects that should be deleted.

Reply:
Done.